src/util/SweptCollection.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+const Collection = require('./Collection.js');
+const { TypeError } = require('../errors/DJSError.js');
+
+/**
+ * Options for defining the behavior of a Swept Collection
+ * @typedef {Object} SweptCollectionOptions
+ * @property {Iterable} [iterable=null] Optional entries passed to the Map constructor.
+ * @property {Client} [client=null] A client that instantiates this and has a setInterval function
+ * @property {number} [maxSize=-1] The maximum size of the Collection
+ * @property {Function|null} [maxSizePredicate=null] A function used to determine which item to remove
+ * when reaching the specified max size. The function takes an entry as a paremeter and returns a boolean.
+ * <info>This can allow the size to grow larger than max size when none of the entries pass the predicate.</info>
+ * @property {Function|null} [sweepFunction=null] A custom function passed directly to `sweep()` every `sweepInterval`.
+ * See {@link [Collection#sweep](https://discord.js.org/#/docs/collection/master/class/Collection?scrollTo=sweep)}
+ * for the definition of this function. This overrides the build in function generated using the other properties.
+ * @property {number} [sweepLifetime=14400] How long an entry should stay in the collection
+ * before it is considered sweepable
+ * @property {string} [sweepLifetimeProperty='createdTimestamp'] The property of the entry to check when
+ * determining the entries lifetime.
+ * @property {boolean} [sweepArchivedOnly=false] Whether to only sweep entries with a property `archived: true`.
+ * E.g. for threads.
+ * @property {number} [sweepInterval=3600] How frequently, in seconds, to sweep the collection.
```

Comment:
So... per hour?

Also, why is this in seconds?

Reply:
Yeah, sweeps hourly by default, I have no idea if thats a good interval, it definitely could be pretty long.

Do you mean in seconds vs minutes or seconds vs ms

Reply:
Seconds vs ms.

Reply:
That is to match the current message sweep options and because sweeping on ms intervals seems impractical, also looks cleaner in end code.